hadoop-ozone/s3gateway/src/main/java/org/apache/hadoop/ozone/s3/endpoint/ObjectEndpoint.java

```diff
@@ -195,16 +211,24 @@ public Response put(
 
       // Normal put object
       OzoneBucket bucket = getBucket(bucketName);
-
-      output = bucket.createKey(keyPath, length, replicationType,
-          replicationFactor, new HashMap<>());
-
       if ("STREAMING-AWS4-HMAC-SHA256-PAYLOAD"
           .equals(headers.getHeaderString("x-amz-content-sha256"))) {
         body = new SignedChunksInputStream(body);
       }
 
-      IOUtils.copy(body, output);
+      if (enableStreaming) {
```

Comment:
Let's move the new code to a new class, say ObjectEndpointStreaming.  Otherwise, there are many changes here and it becomes hard to maintain.
```
  @PUT
  public Response put(
      @PathParam("bucket") String bucketName,
      @PathParam("path") String keyPath,
      @HeaderParam("Content-Length") long length,
      @QueryParam("partNumber")  int partNumber,
      @QueryParam("uploadId") @DefaultValue("") String uploadID,
      InputStream body) throws IOException, OS3Exception {

    if (datastreamEnabled) {
      return objectEndpointStreaming.put(bucketName, keyPath, length, partNumber, uploadID, body);
    }
    ...
  }
```